File: 1817-calculate-money-in-leetcode-bank/1817-calculate-money-in-leetcode-bank.java
class Solution {
    public int totalMoney(int n) {
        int result =0;
        int d = n/7;
        int r = n%7;
        int dollar = 1;
        while(d!=0){
            result+=(7*(dollar+(dollar+6))/2);
            dollar = dollar+1;
            d--;
        }
        System.out.println(dollar);
        System.out.println(r);
        System.out.println(d);
        System.out.println(result);
        while(r!=0){
            result+=dollar;
            dollar++;
            r--;
        } 
        return result;
    }
}

File: 1817-calculate-money-in-leetcode-bank/1817-calculate-money-in-leetcode-bank.rs
impl Solution {
    pub fn total_money(n: i32) -> i32 {
        let mut dollar =1;
        let mut d = n/7;
        let mut r = n%7;
        let mut result =0;
        while(d!=0){
            result += (7*(dollar+(dollar+6))/2);
            d-=1;
            dollar=dollar+1;
        }

        while(r!=0){
            result+=dollar;
            dollar+=1;
            r-=1;
        }
        result
    }
}